pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -115,23 +124,34 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 	if service.Annotations == nil {
 		service.Annotations = map[string]string{}
 	}
-	if infraConfig.Status.Platform == configv1.AWSPlatformType {
+
+	if isProxyProtocolNeeded(ci, infraConfig) {
 		service.Annotations[awsLBProxyProtocolAnnotation] = "*"
-		// Set the load balancer for AWS to be as aggressive as Azure (2 fail @ 5s interval, 2 healthy)
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval"] = "5"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout"] = "4"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-unhealthy-threshold"] = "2"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold"] = "2"
 	}
-	// Azure load balancers are not customizable and are set to (2 fail @ 5s interval, 2 healthy)
-	// GCP load balancers are not customizable and are set to (3 fail @ 8s interval, 1 healthy)
 
-	if infraConfig.Status.Platform == configv1.IBMCloudPlatformType {
-		service.Annotations[iksLBScopeAnnotation] = "public"
+	if infraConfig.Status.PlatformStatus != nil {
+		switch infraConfig.Status.PlatformStatus.Type {
```

Comment:
Use `getPlatformStatus` (see https://github.com/openshift/cluster-ingress-operator/pull/427#discussion_r461210113).  We cannot count on `infraConfig.Status.PlatformStatus` to be set if there is a chance the cluster was upgraded from 4.1.

(Note that the test code is fine—it is safe to assume that the tests are running on a newly installed cluster, or one that has been upgraded from 4.(n-1).)